src/utils/DownloadUtils.ts

```diff
@@ -21,4 +28,66 @@ export function addSolutionsToError(err: R2Error): void {
 export function generateProgressPercentage(progress: number, currentIndex: number, total: number): number {
     const completedProgress = (currentIndex / total) * 100;
     return completedProgress + (progress * 1/total);
-}
\ No newline at end of file
+}
+
```

Comment:
Maybe ponder a bit if placing dependency resolution stuff in DownloadUtils makes sense. Pedantically it's a separate issue from downloading, but sometimes there might be practical limitations that makes being pedantic a bad thing.